Products/ZenUtils/tests/testExceptions.py

```diff
@@ -1,42 +1,67 @@
 #!/usr/bin/env python
 
-from Products.ZenUtils.Exceptions import resolveException, ZentinelException, ZenPathError
-from Products.ZenTestCase.BaseTestCase import BaseTestCase
-
-class Failure(object):
-    def __init__(self):
-        self.value = None
-        self.type = None
-        self.tb = None
-        self.raise_value = None
-
-    def raiseException(self):
-        raise self.raise_value, self.type, self.tb
-
-class ExceptionTestCases(BaseTestCase):
-    def test_resolveException_with_exception_value(self):
-        failure = Failure()
-        failure.value = ZentinelException()
-        self.assertIs( failure.value, resolveException( failure))
-
-    def test_resolveException_with_exception_string(self):
-        failure = Failure()
-        failure.type = "Products.ZenUtils.Exceptions.ZenPathError"
-        self.assertIsInstance( resolveException( failure), ZenPathError)
-
-    def test_resolveException_with_exception_string_with_import_error(self):
-        failure = Failure()
-        failure.value = "value"
-        failure.tb = "tb"
-        failure.type = "Products.ZenUtils.Exceptions.UnknownError"
-        actual = resolveException( failure)
-        self.assertIsInstance( actual, Exception)
-        self.assertEquals( ("value", "tb"), actual.args)
-
-    def test_resolveException_with_raiseException(self):
-        failure = Failure()
-        failure.raise_value = RuntimeError()
-        self.assertIs( failure.raise_value, resolveException( failure))
+from unittest import TestCase
+
+from twisted.spread.pb import RemoteError
+from twisted.python.failure import Failure
+
+from Products.ZenUtils.Exceptions import (
+    resolveException, ZenPathError, ZenResolveExceptionError
+)
+
+
+class ExceptionTestCases(TestCase):
+
+    def setUp(self):
+        self.exception = RuntimeError('genric exception')
+        self.failure = Failure(self.exception)
+
+    def test_resolveException(self):
+        out = resolveException(self.failure)
+        self.assertEqual(out, self.exception)
```

Comment:
This looks to be the only test that uses the artifacts created by `setUp` as is.  The other tests appear to either ignore them entirely or replace all the attributes, effectively making the `setUp` code pointless.  If each test has different setups, a general setup is kind of misleading.